### DISCUSSION TITLE
Credential availability duration

### DISCUSSION THREAD

**Discussion by FilipDem** [bimmerconnected/bimmer_connected#652] [Q&A]
Hi,
I use the bimmer_connected integrated with domoticz (with a python plugin).
Currently I create the account with "MyBMWAccount(...)" at startup and then periodically I use get_vehicles() and get_vehicle(vin) periodically... I got errors at regular basis on the get_vehicles(), however when I restart the plugin with a new login, all works fine...
Is there a need to periodically renew the tokens with a login explicitly? Currently I assumed this was done underneath one of the get_ functions. 
Thanks for some guidance.
Filip

**Comment by rikroe** [comment]
When using `MyBMWAccount` (and keeping it in memory in the python process), then `get_vehicles()` (or any other function that calls the API) will take of logging in/refreshing the token.
However, we don't check it first, we rather "just try" the API and only if there is an issue, we will login again and then try calling the API again.
Calling `account.get_vehicles()` also updates the credentials stored in the account object in memory.

`access_tokens` are valid for one hour, the `refresh_token` used to refresh is valid for much longer. However even if both fail, the code falls back logging in via username/password.

Maybe you also had some fallout of the general login issues during the last days?

**Comment by FilipDem** [comment]
Yes indeed I had also some login troubles recently...
Now it is running since 26/08 (so 3 days) without a problem... I am monitoring further...
Thanks for the explanation already!
